providers/aws-s3/src/main/java/org/jclouds/aws/s3/blobstore/internal/AWSS3PutOptionsToPutObjectOptions.java

```diff
@@ -27,16 +27,45 @@
  * for AWS S3 operations.
  */
 public class AWSS3PutOptionsToPutObjectOptions implements
-Function<PutOptions, PutObjectOptions> {
+Function<PutOptions, AWSS3PutOptionsToPutObjectOptions.ReturnValue> {
+	
+	// Oh look, it's a tuple!  Or at least it would be if, you know, Java didn't have opinions about such things.
+	public static class ReturnValue {
+		
+		private final PutObjectOptions mainOptions;
+		
+		public ReturnValue(boolean useServerSideEncryption) {
+			
+			PutObjectOptions.Builder builder = PutObjectOptions.builder();
+			if (useServerSideEncryption) { builder.serverSideEncryption(); }
+			this.mainOptions = builder.build();
+		}
+
+		public PutObjectOptions getMainOptions() {
+			return mainOptions;
+		}
+
+		/* Part options are always fixed at defaults due to OPSC-7247.  This may change if we start supporting
+		 * additional S3 options (ACLs etc.) in future ops. */
+		public PutObjectOptions getPartOptions() {
+			return PutObjectOptions.DEFAULTS;
+		}
+	}
+
+	/* We can get away with this because yes/no to server-side encryption is the only axis of variation at this point. */
```

Comment:
as in this is the only header that we expect to attach to the requests or this is the only header that can be attached at this point in the code?